File: football-backend/src/main/java/com/testowanie/football/model/Article.java
package com.testowanie.football.model;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.springframework.lang.NonNull;

import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Article extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NonNull
    @Column(nullable = false)
    private String title;

    @NonNull
    @ManyToOne
    private Editor editor;

    @NonNull
    @Column(length = 10000, nullable = false)
    private String content;

    private String photoUrl;

    @NonNull
    @ManyToOne
    private Category category;

    //@NonNull
    @Builder.Default
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Comment> comments = new HashSet<>();
}

File: football-frontend/src/app/models/article.ts
import { Editor } from './editor';
import { Category } from './category';
import { Comment } from './comment';

export class Article {
    constructor(
        public title: string,
        public editor: Editor,
        public publicationDate: Date,
        public content: string,
        public photoUrl: string,
        public category: Category,
        public comments: Comment[],
        public id?: number
    ) {}

    public static fromForm(articleRawFormValue: ArticleRawFormValue): Article {
        return new Article(
            articleRawFormValue.title,
            articleRawFormValue.editor,
            articleRawFormValue.publicationDate,
            articleRawFormValue.content,
            articleRawFormValue.photoUrl,
            articleRawFormValue.category,
            [],
            articleRawFormValue.id ?? undefined
        );
    }
}

export interface ArticleRawFormValue {
    title: string;
    editor: Editor;
    publicationDate: Date;
    content: string;
    photoUrl: string;
    category: Category;
    id?: number;
}
